### ISSUE TITLE
Does not resolve variable

### ISSUE THREAD

**Issue by cg089** [3ative/Making-TTS-Better-NodeRed#1]
Hi,
thank you for the great tips!
Unfortunately, Alexa does not resolve the variables and reads out states.sensor.whatever.sensor.i.try

Do you have any idea why that happens?
I tired any variation of the mustache code. This example flow is expected to output at least once "below_horizon", but does not.

`[{"id":"61f510d9.31bdd","type":"inject","z":"649b1c40.c06774","name":"","topic":"","payload":"","payloadType":"date","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":820,"y":600,"wires":[["aec1d8df.8fe488"]]},{"id":"12fb162d.93d52a","type":"alexa-remote-routine","z":"649b1c40.c06774","name":"Ansage","account":"52f819b8.3ef9b8","routineNode":{"type":"speak","payload":{"type":"regular","text":{"type":"msg","value":"payload"},"devices":["66443902f5ab41e1b9f8b81d28e35988"]}},"x":1200,"y":600,"wires":[[]]},{"id":"aec1d8df.8fe488","type":"function","z":"649b1c40.c06774","name":"","func":"\nmsg.payload =\n\n\"{{sensor.sun.sun}}\" + \n\"{{state.sun.sun.state}}\" + \n\"{{states.sun.sun}}\" + \n\"{{entity.sun.sun}}\"\n\nreturn msg;","outputs":1,"noerr":0,"x":1010,"y":600,"wires":[["12fb162d.93d52a"]]},{"id":"52f819b8.3ef9b8","type":"alexa-remote-account","z":"","name":"","authMethod":"proxy","proxyOwnIp":"192.168.80.2","proxyPort":"3456","cookieFile":"authfile","refreshInterval":"3","alexaServiceHost":"layla.amazon.de","amazonPage":"amazon.de","acceptLanguage":"de-DE","userAgent":"","useWsMqtt":"on","autoInit":"on"}]`

**Comment by 3ative** [comment]
You have a error on your variable.

Use: **"{{states.sun.sun.state}}"** instead

**Comment by cg089** [comment]
Hi,
that does not work either. Alexa is just saying the words states sun sun state. :(
Thanks for answering and for your videos!

**Comment by 3ative** [comment]
Works here, It must be that last node you use... try the one I show in the video.

**Comment by cg089** [comment]
Hi,
of course I tried everything as in the video :)
All the examples are working just fine, but as soon as a variable comes in, it does not resolve the variable but reads it out word for word.
I ended up just trying different kind of variables (and of course different sensors that work fine in other flows) in my example to find out what's the error, but I can't resolve any of the different tryings.
Thanks anyway!
